controllers/storagecluster/provider_server.go

```diff
@@ -155,6 +157,16 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
 	}
 
 	r.Log.Info("Service create/update succeeded")
+
+	nodeAddresses, err := r.GetWorkerNodesInternalIPAddresses()
+	if err != nil {
+		return err
+	}
+
+	instance.Status.StorageProviderEndpoint = nodeAddresses[0] + ":" + fmt.Sprint(actualService.Spec.Ports[0].NodePort)
```

Comment:
I think it is a bad idea to access the array by constant index. Do we give the port a name? if so we can search by name

Reply:
made changes on both